### ISSUE TITLE
Fix Hugo warnings

### ISSUE THREAD

**Issue by yurishkuro** [jaegertracing/documentation#669]
We have warnings during the build of the site, some of which indicate they may turn into failures in the future due to deprecation.  I counted at least 3 different warnings:
```
make develop
...

WARN  DEPRECATED: Kind "taxonomyterm" used in disableKinds is deprecated, use "taxonomy" instead.
...

WARN  .File.Path on zero object. Wrap it in if or with: {{ with .File }}{{ .Path }}{{ end }}
WARN  .Path when the page is backed by a file is deprecated. We plan to use Path for a canonical source path and you probably want to check the source is a file. To get the current behaviour, you can use a construct similar to the one below:

  {{ $path := "" }}
  {{ with .File }}
	{{ $path = .Path }}
  {{ else }}
	{{ $path = .Path }}
  {{ end }}
...

WARN  found no layout file for "html" for kind "section": You should create a template file which matches Hugo Layouts Lookup Rules for this combination.
```

**Comment by tico88612** [comment]
Hi @yurishkuro

I want to confirm the Hugo build version; currently, I am following Netlify CI's Hugo 0.102.3, and I confirm that some warnings exist (except for `WARN  DEPRECATED: Kind "taxonomyterm" used in disableKinds is deprecated, use "taxonomy" instead.`, which does not appear in Hugo 0.102.3).

After confirming the version, I can try to fix the warning.

**Comment by yurishkuro** [comment]
```
$ grep -i version netlify.toml
HUGO_VERSION = "0.102.3"
```

**Comment by yurishkuro** [comment]
my local version is the same+extended
```
$ hugo version
hugo v0.120.3-a4892a07b41b7b3f1f143140ee4ec0a9a5cf3970+extended darwin/arm64 BuildDate=2023-11-01T17:57:00Z VendorInfo=brew
```

**Comment by tico88612** [comment]
> my local version is the same+extended
> 
> ```
> $ hugo version
> hugo v0.120.3-a4892a07b41b7b3f1f143140ee4ec0a9a5cf3970+extended darwin/arm64 BuildDate=2023-11-01T17:57:00Z VendorInfo=brew
> ```

exactly not the same; your version is `v0.120.3`, CI version is `v0.102.3`.

**Comment by yurishkuro** [comment]
ok, well we should make it work for the latest version anyway, since it's not going to get better

**Comment by tico88612** [comment]
Now Hugo latest version is `0.123.6`, should we follow this version?

**Comment by yurishkuro** [comment]
Sure

**Comment by tico88612** [comment]
Hi @yurishkuro 

#675
I fixed all warning & error in Hugo 0.123.6. Please feel free to review it.
